crates/anemo/src/network/mod.rs

```diff
@@ -134,11 +137,42 @@ impl Builder {
             .alternate_server_name(alternate_server_name)
             .private_key(private_key)
             .build()?;
-        let socket = std::net::UdpSocket::bind(self.bind_address)?;
-        let endpoint = Endpoint::new(endpoint_config, socket)?;
 
-        let config = Arc::new(config);
+        let addrs: Vec<_> = self.bind_address.to_socket_addrs()?.collect();
+        let socket = Socket::new(
+            Domain::for_address(
+                *addrs
+                    .first()
+                    .ok_or_else(|| anyhow!("could not resolve address {:?}", self.bind_address))?,
+            ),
+            Type::DGRAM,
+            Some(Protocol::UDP),
+        )?;
+        for addr in addrs {
+            socket.bind(&socket2::SockAddr::from(addr))?;
+        }
+        if let Some(send_buffer_size) = quic_config.socket_send_buffer_size {
+            socket.set_send_buffer_size(send_buffer_size)?;
```

Comment:
FYI this may require a change in mysten-sim to ignore `SO_SNDBUF` in the setsockopt call.